### ISSUE TITLE
Add emulators for cluster mass function

### ISSUE THREAD

**Issue by elisachisari** [LSSTDESC/CCL#235]
This was a suggestion from the cluster experts, pointing out that the Tinker mass function currently implemented is not accurate enough for LSST. What do you think, @tmcclintock?

**Comment by villarrealas** [comment]
I agree. Tinker is not calibrated with sufficient accuracy on those scales. It is not obvious to me that emulators necessarily solves this though - getting sufficient statistics at that mass scale is a little expensive.

**Comment by salmanhabib** [comment]
It's not accurate enough, but it's a reference model that's used a lot. 
LSST needs roughly a factor of 5-10 improvement in accuracy.

**Comment by salmanhabib** [comment]
We have simulations at large enough volumes. The accuracy requirements 
are less stringent at high mass ("clusters") and become tighter at lower 
masses ("groups") due to statistics. This has been studied in a number 
of papers.

**Comment by villarrealas** [comment]
Makes sense to me. Getting an emulator implemented for hmf calculation was always an end goal. Perhaps it should be made the default over Tinker once implemented though.

**Comment by tmcclintock** [comment]
It's probably premature to make anything the default over Tinker. Even though @SebastianBocquet 's emulator and mine will almost certainly work better, I think that they should at least be used in some kind of analysis before becoming the default in CCL.

**Comment by villarrealas** [comment]
I worry about leaving something as the default option when it is insufficient at scales that will be used. Though your point is true: Tinker has the advantage of being widely used in the literature to date.

**Comment by tmcclintock** [comment]
It will still be a while until it is "outdated". I think it's fine for now. More importantly it's much easier to compare it with existing codes, which is of more immediate importance for this project.

**Comment by SebastianBocquet** [comment]
I think there are two issues here, that we should treat and discuss separately: i) the science question whether the Tinker HMF is accurate enough for LSST (it isn't, and remember it is not just about box size, but also about the cosmological dependence!) and whether our HMF emulators will improve on it (yes, they will, and it's our job to show this in the papers we're writing. At that point, I'd claim that Tinker will be outdated.), and ii) what should go into CCL. I think CCL should reflect the current state of the art, but only contain published work. So, as soon as the emulators are published (and confirmed to be superior to the Tinker HMF), they should be implemented as the default.

**Comment by villarrealas** [comment]
This probably raises the repeating issue of validation and inclusion of outside packages.

**Comment by SebastianBocquet** [comment]
Yes, but this is true for the Tinker HMF as well, where you assume that Tinker's parameters are correct.

**Comment by villarrealas** [comment]
Agreed. We really only validated that we reproduce Tinker to high precision, but a lot of hmf choices disagree at the 5-10% level. So the question of validation does need to be clarified by us on a policy level. What does it mean for the emulator to be validated according to CCL, effectively.

**Comment by salmanhabib** [comment]
This is a complicated question. How can you verify/validate packages 
that are based on years of work by other people using complicated codes? 
Frankly, I don't think it's possible. All one can do is refer to the 
papers and assume their error estimates are (more or less) correct, 
unless proven otherwise.

**Comment by villarrealas** [comment]
I am inclined to agree. It is a different beast than our previous validation. Whatever our decision though, it should be codified, though.

**Comment by c-d-leonard** [comment]
@damonge was this addressed at all in your recent halo theory merges?

**Comment by damonge** [comment]
Nope
